### PR TITLE
introducing unit test for rucio copytool

### PR DESCRIPTION
**Pull request by TomasJavurek**

### CODE REVIEW

File: pilot/copytool/rucio.py [hunk 1]
```diff
@@ -44,6 +46,7 @@ def copy_in(files, **kwargs):
         :raise: PilotException in case of controlled error
     """
 
+    trace_object = TraceReport()
```
Comment (anisyonk) [inline pilot/copytool/rucio.py]: Introducing new `TraceReport` instance  `trace_object`  is redundant and actually useless, why do you need it?

there is already properly filled instance of `TraceReport` available via passed  `trace_report`  variable to a copytool module by top workflow

**Comment by anisyonk** [comment]
Tomas,  your changes overwrites/lost the master fixes,
either you got a mess with git hard reset or did overwrite them manually.
can you please check.

File: pilot/copytool/rucio.py (hunk 1 above)
Comment (TomasJavurek) [reply pilot/copytool/rucio.py]: trace_report was a string, at least in previous versions, not the TraceReport() object.

**Comment by TomasJavurek** [comment]
Yes, I got mess with git hard reset. This is usually used in Rucio. I have no clue, how to get out of the mess.

File: pilot/copytool/rucio.py (hunk 1 above)
Comment (TomasJavurek) [reply pilot/copytool/rucio.py]: sorry, trace_report is dict, not string. However, it is not TraceReport object.

Comment (anisyonk) [reply pilot/copytool/rucio.py]: sorry, but you in the tests initialized `trace_report` as a dictionary:)

https://github.com/PanDAWMS/pilot2/compare/91fd4976d283e0c50c67c88f40b5fdbbde060cb9..4e8567e17b2ff4efc484d771e59ace73a2de3f98#diff-f722e8348b3c3f6c0af1560646e475e2L124

Comment (TomasJavurek) [reply pilot/copytool/rucio.py]: sorry, stupid mistake

File: pilot/test/test_copytools_rucio.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        infiles = ""
+        fsize = ""
+        realdatasetsin = ""
+        guid = ""
+        checksum = ""
+        scope = ""
+        ddmendpointin = ""
+        turl = ""
+        """ Create temp files in source dir """
+        for i in range(0, self.numFiles):
```
[the remark this answers is not in the document]
Comment (TomasJavurek) [reply pilot/test/test_copytools_rucio.py]: I deleted that already in the last pull request.

**Pull request closed without merging**